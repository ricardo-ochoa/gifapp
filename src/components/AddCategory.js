import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const  AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSumit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit = { handleSumit }>
            <h2>Add Category</h2>
            <input
                type='text'
                value={ inputValue }
                onChange={ handleInputChange }
            ></input>
        </form>
    )
}

AddCategory.prototypes = {
    setCategories: PropTypes.func.isRequired
}