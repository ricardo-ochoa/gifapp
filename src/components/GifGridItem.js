import React from 'react'
import PropTypes from 'prop-types';


export const GifGridItem = ({id, url, title }) => {


    return (
        <>
            <div className="card animate__animated animate__pulse">
                <img src= {url} alt={ title }></img>
                      
            </div>
            
        </>
    )
}

GifGridItem.prototypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
