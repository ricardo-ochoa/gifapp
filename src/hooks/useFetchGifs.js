import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    })


    //useEffect ayuda a que se ejecute una vez cuando se renderiza el componente.
    useEffect(() => {
        getGifs( category )
            .then( imgs => {
                setstate({
                    data: imgs,
                    loading: false
                });
            });
    }, [ category ])   

    return state;
}
