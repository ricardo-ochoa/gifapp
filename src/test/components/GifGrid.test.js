import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';

//Falsear la data para probar el Hook useFetchGifs
jest.mock('../../hooks/useFetchGifs');

describe('Debe mostrar Gifgrid correctamente', () => {

    const category = "One Punch Man";

    test('Debe de mostrar toMatchSnapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />) 
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />) 
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe(false);
    }) 
})
