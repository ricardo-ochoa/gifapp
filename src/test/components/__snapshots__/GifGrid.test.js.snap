// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe mostrar Gifgrid correctamente Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Debe mostrar Gifgrid correctamente Debe de mostrar toMatchSnapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
